cli/internal/cmd/applyinit.go

```diff
@@ -0,0 +1,238 @@
+	req := &initproto.InitRequest{
+		KmsUri:               masterSecret.EncodeToURI(),
+		StorageUri:           uri.NoStoreURI,
+		MeasurementSalt:      measurementSalt,
+		KubernetesVersion:    versions.VersionConfigs[conf.KubernetesVersion].ClusterVersion,
+		KubernetesComponents: versions.VersionConfigs[conf.KubernetesVersion].KubernetesComponents.ToInitProto(),
+		ConformanceMode:      a.flags.conformance,
+		InitSecret:           stateFile.Infrastructure.InitSecret,
+		ClusterName:          stateFile.Infrastructure.Name,
+		ApiserverCertSans:    stateFile.Infrastructure.APIServerCertSANs,
+	}
+	a.log.Debugf("Sending initialization request")
+	resp, err := a.initCall(cmd.Context(), a.newDialer(validator), stateFile.Infrastructure.ClusterEndpoint, req)
+	a.spinner.Stop()
+	a.log.Debugf("Initialization request finished")
+
+	if err != nil {
+		var nonRetriable *nonRetriableError
+		if errors.As(err, &nonRetriable) {
+			cmd.PrintErrln("Cluster initialization failed. This error is not recoverable.")
+			cmd.PrintErrln("Terminate your cluster and try again.")
+			if nonRetriable.logCollectionErr != nil {
+				cmd.PrintErrf("Failed to collect logs from bootstrapper: %s\n", nonRetriable.logCollectionErr)
+			} else {
+				cmd.PrintErrf("Fetched bootstrapper logs are stored in %q\n", a.flags.pathPrefixer.PrefixPrintablePath(constants.ErrorLog))
+			}
+		}
+		return nil, err
+	}
+	a.log.Debugf("Initialization request successful")
+
+	a.log.Debugf("Buffering init success message")
+	bufferedOutput := &bytes.Buffer{}
+	if err := a.writeOutput(stateFile, resp, a.flags.mergeConfigs, bufferedOutput, measurementSalt); err != nil {
+		return nil, err
+	}
+
+	return bufferedOutput, nil
+}
+
+// initCall performs the gRPC call to the bootstrapper to initialize the cluster.
+func (a *applyCmd) initCall(ctx context.Context, dialer grpcDialer, ip string, req *initproto.InitRequest) (*initproto.InitSuccessResponse, error) {
+	doer := &initDoer{
+		dialer:   dialer,
+		endpoint: net.JoinHostPort(ip, strconv.Itoa(constants.BootstrapperPort)),
+		req:      req,
+		log:      a.log,
+		spinner:  a.spinner,
+		fh:       file.NewHandler(afero.NewOsFs()),
+	}
+
+	// Create a wrapper function that allows logging any returned error from the retrier before checking if it's the expected retriable one.
+	serviceIsUnavailable := func(err error) bool {
+		isServiceUnavailable := grpcRetry.ServiceIsUnavailable(err)
+		a.log.Debugf("Encountered error (retriable: %t): %s", isServiceUnavailable, err)
+		return isServiceUnavailable
+	}
+
+	a.log.Debugf("Making initialization call, doer is %+v", doer)
+	retrier := retry.NewIntervalRetrier(doer, 30*time.Second, serviceIsUnavailable)
+	if err := retrier.Do(ctx); err != nil {
+		return nil, err
+	}
+	return doer.resp, nil
+}
+
+// generateMasterSecret reads a base64 encoded master secret from file or generates a new 32 byte secret.
+func (a *applyCmd) generateMasterSecret(outWriter io.Writer) (uri.MasterSecret, error) {
```

Comment:
nit: since it also stores a file the name should reflect that.
```suggestion
func (a *applyCmd) generateAndPersistMasterSecret(outWriter io.Writer) (uri.MasterSecret, error) {
```